drivers/iio/dac/ad5413.c

```diff
@@ -0,0 +1,761 @@
+    ret = ad5413_find_out_range(st, range, size,
+                                tmparray[0], tmparray[1]);
+    if (ret) {
+        dev_err(&st->spi->dev, "Invalid range values (%u, %u)\n",
+                tmparray[0], tmparray[1]);
+        return ret;
+    }
+
+    /* 3) read slew-time（optional） */
+    ret = device_property_read_u32(&st->spi->dev,
+                                  "adi,slew-time-us", &tmp);
+    if (ret) {
+        dev_dbg(&st->spi->dev, "No \"adi,slew-time-us\" property, disabling slew\n");
+        st->slew_time = 0;
+    } else {
+        st->slew_time = tmp;
+    }
+
+    return 0;}
+
+static int ad5413_init(struct ad5413_state *st)
+{
+	int regval, ret;
+
+	st->gpio_reset = devm_gpiod_get_optional(&st->spi->dev, "reset",
+						 GPIOD_OUT_HIGH);
+	if (IS_ERR(st->gpio_reset))
+		return PTR_ERR(st->gpio_reset);
+
+	/* Disable CRC checks */
+	ret = ad5413_crc_disable(st);
+	if (ret < 0)
+		return ret;
+
+	/* Perform a reset */
+	ret = ad5413_reset(st);
+	if (ret < 0)
+		return ret;
+
+	/* Disable CRC checks */
+	ret = ad5413_crc_disable(st);
+	if (ret < 0)
+		return ret;
+
+	/* Perform a calibration memory refresh */
+	ret = ad5413_calib_mem_refresh(st);
+	if (ret < 0)
+		return ret;
+
+	regval = ad5413_spi_reg_read(st, AD5413_DIGITAL_DIAG_RESULTS);
+	if (regval < 0)
+		return regval;
+
+	/* Clear all the error flags */
+	ret = ad5413_spi_reg_write(st, AD5413_DIGITAL_DIAG_RESULTS, regval);
+	if (ret < 0)
+		return ret;
+
+	/* Configure the output range */
+	ret = ad5413_set_out_range(st, st->out_range.reg);
+	if (ret < 0)
+		return ret;
+
+	/* Enable Slew Rate Control, set the slew rate clock and step */
+	if (st->slew_time) {
+		ret = ad5413_slew_rate_config(st);
+		if (ret < 0)
+			return ret;
+	}
+
+	/* Power up the DAC and internal (INT) amplifiers */
+	ret = ad5413_internal_buffers_en(st, 1);
+	if (ret < 0)
+		return ret;
+
+	/* Enable VIOUT */
+	return ad5413_spi_write_mask(st, AD5413_DAC_CONFIG,
+				     AD5413_DAC_CONFIG_OUT_EN_MSK,
+				     AD5413_DAC_CONFIG_OUT_EN_MODE(1));
+}
+
+static int ad5413_probe(struct spi_device *spi)
+{
+	struct ad5413_state *st;
+	struct iio_dev *indio_dev;
+	int ret;
+
+	indio_dev = devm_iio_device_alloc(&spi->dev, sizeof(*st));
+	if (!indio_dev)
+		return -ENOMEM;
+
+	st = iio_priv(indio_dev);
+	spi_set_drvdata(spi, indio_dev);
+
+	st->spi = spi;
+
+	mutex_init(&st->lock);
+
+	indio_dev->name = spi_get_device_id(spi)->name;
+	indio_dev->info = &ad5413_info;
+	indio_dev->modes = INDIO_DIRECT_MODE;
+	indio_dev->num_channels = 1;
+
+	ret = ad5413_parse_dt(st);
+	if (ret < 0)
+		return ret;
+    
+    /* By output range, decide voltage or current channel */
+    if (st->out_range.reg == AD5413_RANGE_PLUSMINUS_10_5V)
+        indio_dev->channels = ad5413_voltage_ch;
+    else
+        indio_dev->channels = ad5413_current_ch;
+    
+	ret = ad5413_init(st);
+	if (ret < 0) {
+		dev_err(&spi->dev, "AD5413 init failed\n");
+		return ret;
+	}
```

Comment:
```suggestion
	if (ret < 0)
		return dev_err_prove(&spi->dev, ret, "AD5413 init failed\n");
```

This applies to all uses of `dev_err` in functions called only during probe.